### PR TITLE
Re-work deployer logic to use existing services

### PR DESCRIPTION
**Pull request by stevemar**
The deployer currently always creates a free tier conversation
service and uploads a new workspace to that service. That leads to
many services and being unable to find their newly uploaded
workspace from the devops pipeline (it links to whatever it finds
first in the backend).

The idea here is to create a new service only if one doesn't already
exist. If one exists, use it. If more than 1 exists, then use the
first found, this isn't any worse than the current situation.

### CODE REVIEW

**Comment by stevemar** [comment]
This would close issue #8

File: .bluemix/pipeline.yml [hunk 1]
```diff
@@ -27,19 +27,46 @@ stages:
...
+      if [ "$result" -eq 0 ]; then
+        cf create-service conversation free "$CHATBOT_NAME"
         cf create-service-key "$CHATBOT_NAME" "$CHATBOT_NAME"
+      # if more than one exists, issue a warning
+      elif [ "$result" -gt 1 ]; then
+        echo "found more than one conversation service, using the first found"
+        # TODO: Use the paid service over the free?
       fi
+      # Finally, we get the service name, either newly created or first result
+      service_name=$(cf services | grep conversation | cut -d " " -f1)
```
Comment (stevemar) [inline .bluemix/pipeline.yml]: unfortunately, this won't work if $CHATBOT_NAME has a space... hmm...

File: .bluemix/pipeline.yml [hunk 2]
```diff
@@ -27,19 +27,46 @@ stages:
...
+      # Chatbot-20170822164726835   conversation          free                create succeeded
+      # Conversation-hf             conversation          free                create succeeded
+      # NOTE: the -c option for grep counts number of rows
+      result=$(cf services | grep -c conversation)
 
-      # Also create similarly named service key / credential for the service
-      if ! cf service-key $CHATBOT_NAME $CHATBOT_NAME >/dev/null; then
+      # if no consersation exists, create service and service key, name after the chatbot
+      if [ "$result" -eq 0 ]; then
+        cf create-service conversation free "$CHATBOT_NAME"
```
Comment (markstur) [inline .bluemix/pipeline.yml]: How about using a known name or prefix? Naming the shared service after the first chatbot doesn't seem to make sense anymore. 1 name sounds easy. Prefix would be to allow more services for future need to avoid too many conversations per service. E.g. name prefix + number or datetime

Comment (stevemar) [reply .bluemix/pipeline.yml]: I went with BAE+datetime

File: .bluemix/pipeline.yml (hunk 1 above)
Comment (stevemar) [reply .bluemix/pipeline.yml]: spoke with the CF folks in their #cli slack, and they didn't provide a better solution. though with mark's suggestion we are a better shape since the space was coming from the chatbot name, mostly. if the user creates the service via BAE then it won't have a space (it'll be BAE+datetime), and through the UI it doesn't have one by default. So though it's possible to create a service with a space, it's unlikely (famous last words)

**Review by markstur — APPROVED** [review]

**Comment by rhagarty** [comment]
We can merge to test it out. I don't think we have solved all of the corner cases, like BAE instance exists but it is the second instance returned. Now we have BAE chatbot workspaces added to the non-BAE conversation instance. That could be confusing.
